### PR TITLE
(403) Add Prison client

### PR DESCRIPTION
**Pull request by Gweaton**
## Context

https://trello.com/c/d4yEqtSO/403-fetch-prison-information

## Changes in this PR

This adds a Prison client for fetching Prison data from the Prisons API. We'll use it to populate the "Offerings" page in the near future. It also starts up the Prisons API locally as a backing service.

There's still an ongoing conversation as to whether this is the correct API to use, but it does seem to have what we need, for now at least. We'll wrangle the Prison object we get back into our own internal `Organisation` model in future work.

### CODE REVIEW

File: server/data/prisonClient.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import RestClient from './restClient'
+import { Prison } from '../@types/prisonApi'
+import config, { ApiConfig } from '../config'
+import paths from '../paths/prisonApi'
+
+export default class PrisonClient {
+  restClient: RestClient
+
+  constructor(token: string) {
+    this.restClient = new RestClient('programmeClient', config.apis.prisonApi as ApiConfig, token)
```
Comment (yndajas) [inline server/data/prisonClient.ts:10]: `'prisonClient'`

**Review by yndajas — COMMENTED** [review]
Looks good. Just a few small things. Also, should the prisons API be accessible on 9091 when running `script/server`? I'm not seeing it

File: server/data/prisonClient.ts (hunk 1 above)
Comment (Gweaton) [reply server/data/prisonClient.ts]: oops!

File: server/testutils/factories/prison.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      {
+        premise,
+        street: streetAddress,
+        locality: county,
+        town: city,
+        postalCode: faker.location.zipCode(),
+        country,
+        primary: true,
+      },
+    ],
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply server/testutils/factories/prison.ts]: I've had a go a this - let me know what you think

**Review by yndajas — COMMENTED** [review]
A few more queries and minor things to clean up!

File: server/testutils/factories/prisonAddress.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  return {
+    premise,
+    street: faker.location.streetAddress(),
+    locality,
+    town: faker.location.city(),
+    postalCode: faker.location.zipCode(),
+    country: faker.location.country(),
+    primary: true,
+  }
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply server/testutils/factories/prisonAddress.ts]: Fishery does this for us, so it uses the value pass in as a parameter, or if one isn't passed in, it uses the value you've defined in the factory.

File: server/testutils/factories/prisonAddress.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { faker } from '@faker-js/faker'
```
Comment (yndajas) [inline server/testutils/factories/prisonAddress.ts:1]: We can use `@faker-js/faker/locale/en_GB` to get proper UK postcodes (and more UK-style data in general). Can do the same in `prison`, but the post code is where it most noticeably affects the format

**Review by yndajas — APPROVED** [review]
Looks good. Just one minor thing that might be good to update then I think it's ready to go

File: server/testutils/factories/prisonAddress.ts (hunk 4 above)
Comment (Gweaton) [reply server/testutils/factories/prisonAddress.ts]: Good spot, have added this and hardcoded 'United Kingdom' as it didn't make sense for things like "Uganda" to show with a UK postcode. We can revisit this in future if we need but I don't think we'll need to go more granular in our test data just yet.

**Pull request merged**